network/peer.go

```diff
@@ -836,3 +868,50 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time().Unix(),
+	})
+}
+
+// assumes stateLock is held
+func (p *peer) releaseAliases() {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	for _, alias := range p.aliases {
+		str := alias.ip.String()
+		delete(p.net.peerAliasIPs, str)
+
+		p.net.log.Verbo("released alias %s for peer %s", str, p.id.String())
+	}
+	p.aliases = p.aliases[:0]
+}
+
+// assumes stateLock is not held
+func (p *peer) releaseExpiredAliases() {
```

Comment:
Should we combine this method with `releaseAliases`?
It could take a `time.Time` as an argument and release all aliases recorded before that time.

Reply:
Thought this would be cumbersome because we'd have to communicate whether or not `stateLock` was held but I played around a little and it looks much better.